blenderproc/python/loader/URDFLoader.py

```diff
@@ -0,0 +1,454 @@
+                      `urdf_object.set_location_ik()`) are being handled internally. Useful for visualization in
+                      blender.
+    :return: URDF object instance.
+    """
+    # install urdfpy
+    SetupUtility.setup_pip(user_required_packages=["git+https://github.com/wboerdijk/urdfpy.git"])
+    from urdfpy import URDF
+
+    # load urdf tree representation
+    urdf_tree = URDF.load(urdf_file)
+
+    # create new empty armature
+    bpy.ops.object.select_all(action='DESELECT')
+    bpy.ops.object.armature_add()
+    armature = bpy.context.active_object
+    armature.name = urdf_tree.name
+    # remove initially created bone
+    bpy.ops.object.mode_set(mode='EDIT')
+    armature.data.edit_bones.remove(armature.data.edit_bones.values()[0])
+    bpy.ops.object.mode_set(mode='OBJECT')
+
+    # recursively create bones, starting at the base bone(s)
+    base_joints = get_joints_which_have_link_as_parent(urdf_tree.base_link.name, urdf_tree.joints)
+    for base_joint in base_joints:
+        create_bone(armature, base_joint, urdf_tree.joints, parent_bone_name=None, create_recursive=True,
+                    fk_offset=fk_offset, ik_offset=ik_offset)
+
+    # load links
+    links = load_links(urdf_tree.links, urdf_tree.joints, armature, urdf_path=urdf_file)
+
+    # propagate poses through the links
+    for base_joint in base_joints:
+        propagate_pose(links, base_joint, urdf_tree.joints, armature)
+
+    # parent links with the bone
+    for link in links:
+        link.parent_with_bone(weight_distribution=weight_distribution)
+
+    # set to forward kinematics per default
+    for link in links:
+        link.switch_fk_ik_mode(mode="fk")
+
+    urdf_object = URDFObject(armature, links=links, xml_tree=urdf_tree)
+
+    # hide all inertial and collision objects per default
+    urdf_object.hide_irrelevant_objs()
+
+    return urdf_object
+
+
+def get_joints_which_have_link_as_parent(link_name: str, joint_trees: List["urdfpy.Joint"]) -> List["urdfpy.Joint"]:
+    """ Returns a list of joints which have a specific link as parent.
+
+    :param link_name: Name of the link.
+    :param joint_trees: List of urdfpy.Joint objects.
+    :return: List of urdfpy.Joint objects.
+    """
+    return [joint_tree for i, joint_tree in enumerate(joint_trees) if joint_tree.parent == link_name]
+
+
+def get_joints_which_have_link_as_child(link_name: str, joint_trees: List["urdfpy.Joint"]) -> Union["urdfpy.Joint", None]:
+    """ Returns the joint which is the parent of a specific link.
+
+    :param link_name: Name of the link.
+    :param joint_trees: List of urdfpy.Joint objects.
+    :return: List of urdfpy.Joint objects, or None if no joint is defined as parent for the respective link.
+    """
+    valid_joint_trees = [joint_tree for i, joint_tree in enumerate(joint_trees) if joint_tree.child == link_name]
+    if valid_joint_trees == []:  # happens for the very first link
```

Comment:
```suggestion
    if not valid_joint_trees:  # happens for the very first link
```

correct check for if a list empty